specs/lib/gltfPipelineSpec.js

```diff
@@ -104,22 +104,46 @@ describe('gltfPipeline', function() {
             callback();
         });
         var options = {
-            createDirectory : false
+            createDirectory : false,
         };
         readGltf(gltfPath, options, function(gltf) {
+            var options = { basePath : path.dirname(gltfPath) };
             processJSONToDisk(gltf, outputPath, options, function() {
-                expect(path.normalize(spy.calls.first().args[0])).toEqual(path.normalize('./output/'));
+                expect(path.normalize(spy.calls.first().args[0])).toEqual(path.normalize('output/output'));
+                done();
+            });
+        });
+    });
+
+    it('will write sources from JSON', function(done) {
+        var options = {};
+        readGltf(gltfPath, options, function (gltf) {
+            var initialSource = gltf['buffers'].CesiumTexturedBoxTest.extras._pipeline.source;
+            processJSON(gltf, options, function () {
+                var finalSource = gltf['buffers'].CesiumTexturedBoxTest.extras._pipeline.source;
+                expect(initialSource).not.toEqual(finalSource);
+                done();
```

Comment:
For these two tests it's more accurate to compare the data uris since we are checking that it wrote the data back to those.